src/main/java/com/taskadapter/redmineapi/AttachmentManager.java

```diff
@@ -110,7 +110,41 @@ public Attachment uploadAttachment(String contentType, File content)
         }
     }
 
+/**
+	 * Uploads an attachment.
+	 * 
+	 * @param fileName
+	 *            file name of the attachment.
+	 * @param contentType
+	 *            content type of the attachment.
+	 * @param content
+	 *            attachment content stream.
+	 * @return attachment content.
+	 * @throws RedmineException
+	 *             if something goes wrong.
+	 * @throws IOException
+	 *             if input cannot be read. This exception cannot be thrown yet (I am not sure if http client can distinguish "network" errors and local errors) but is will be good to distinguish
+	 *             reading errors and transport errors.
+	 */
+	public Attachment uploadAttachment(File file, String contentType) throws RedmineException, IOException {
+		final String token;
+		try {
+			token = transport.upload(file);
+			final Attachment result = AttachmentFactory.create();
+			result.setToken(token);
+			result.setContentType(contentType);
+			result.setFileName(file.getName());
+			return result;
+		} catch (RedmineException e) {
+			unwrapException(e, "uploadStream");
+			throw e;
+		}
+	}
+
     /**
+     * 
+     * @decrepated
```

Comment:
And what is a suggested altervative for this deprecated method? In some cases I (or other users) do not have a file to upload. They just have a content. For example, it could be an in-memory image taken from the mobile camera. And the library user do not want to ask for "file" permission for his application. Same for applets on the page. Some generated data.

These use-cases are valid. I do not see any reasons to drop support for them. So if you want to deprecate this method, you have to provide a viable alternative. It will be good if alternative do not cause any inconvenience (like a need to create an external file).